example/mavlink_ftp_client/mavlink_ftp_client.cpp

```diff
@@ -0,0 +1,399 @@
+/**
+ * @file mavlink_ftp.cpp
+ *
+ * @brief Demonstrates how to use Mavlink FTP client with the MAVSDK.
+ *
+ * @author Matej Frančeškin <matej@auterion.com>,
+ * @date 2019-09-06
+ */
+
+#include <mavsdk/mavsdk.h>
+#include <mavsdk/plugins/mavlink_ftp/mavlink_ftp.h>
+
+#include <functional>
+#include <future>
+#include <iostream>
+#include <iomanip>
+#include <cstring>
+
+#define ERROR_CONSOLE_TEXT "\033[31m" // Turn text on console red
+#define NORMAL_CONSOLE_TEXT "\033[0m" // Restore normal console colour
+
+using namespace mavsdk;
+
+void usage(const std::string& bin_name)
+{
+    std::cout << NORMAL_CONSOLE_TEXT << "Usage : " << bin_name
+              << " <connection_url> <server component id> <command> <parameters>" << std::endl
+              << std::endl
+              << "Connection URL format should be :" << std::endl
+              << " For TCP : tcp://[server_host][:server_port]" << std::endl
+              << " For UDP : udp://[bind_host][:bind_port]" << std::endl
+              << " For Serial : serial:///path/to/serial/dev[:baudrate]" << std::endl
+              << "For example, to connect to the simulator use URL: udp://:14540" << std::endl
+              << std::endl
+              << "Server component id is for example 1 for autopilot or 99 for companion computer."
+              << std::endl
+              << std::endl
+              << "Supported commands :" << std::endl
+              << "put <file> <path>    : Upload one file to remote directory" << std::endl
+              << "get <file> <path>    : Download remote file to local directory" << std::endl
+              << "delete <file>        : Delete remote file" << std::endl
+              << "rename <old> <new>   : Rename file" << std::endl
+              << "dir <path>           : List contents of remote directory" << std::endl
+              << "mkdir <path>         : Make directory on remote machine" << std::endl
+              << "rmdir <path>         : Remove directory on remote machine" << std::endl
+              << "cmp <local> <remote> : Compare local and remote file" << std::endl
+              << "crc32 <path>         : Get remote file crc32" << std::endl
+              << "localcrc32 <path>    : Get local file crc32" << std::endl;
```

Comment:
@MatejFranceskin a couple of things:
- can you generally document what the exit code is?
- mkdir: if the directory already exists, an error EEXIST errno is returned from the server. Adding an argument to ignore that would help, or at least indicate it in the exit code.
- similar for delete/rmdir: I need a way to differentiate/ignore if it does not exist
- rmdir: it's not recursive. A recursive variant would be helpful (I'll need it)
- rename: this also works for directories if they are on the same file system